File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.py
# Last updated: 7/13/2026, 10:17:37 AM
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s=="":
            return 0
        else:
            st=0
            e=0
            maxlength=0
            uc=[]
            while e<len(s):
                if s[e] not in uc:
                    uc.append(s[e])
                    e+=1
                    maxlength=max(maxlength,len(uc))
                else:
                    uc.remove(s[st])
                    st+=1
            return maxlength

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 7/13/2026, 10:17:25 AM
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> set= new HashSet<>();
        int l=0,r=0,maxl=0;
        while(r<s.length()){
            char c=s.charAt(r);
            if(!set.contains(c)){
                set.add(c);
                maxl=Math.max(maxl,r-l+1);
                r++;
            }else{
                set.remove(s.charAt(l));
                l++;
            }
        }
    return maxl;
    }
}